src/main/resources/db/changelog/changes/v1.0.4/sql/derived-tables/create-view-purchase-order-lines.sql

```diff
@@ -0,0 +1,56 @@
+    o.jsonb ->> 'approved'::text AS  purchase_order_approved,
+    o.jsonb ->> 'orderType'::text AS  purchase_order_type,
+    o.jsonb ->> 'poNumber'::text AS po_number,
+    o.jsonb ->> 'workflowStatus'::text AS  purchase_order_workflow_status,
+    (o.jsonb -> 'metadata'::text) ->> 'createdDate'::text AS  purchase_order_created_date,
+    (o.jsonb -> 'metadata'::text) ->> 'createdByUserId'::text AS  purchase_order_created_by_id,
+	concat((user_details.jsonb -> 'personal'::text) ->> 'lastName'::text, ', ',
+              (user_details.jsonb -> 'personal'::text) ->> 'firstName'::text) AS  purchase_order_created_by,
+    (o.jsonb -> 'metadata'::text) ->> 'updatedDate'::text AS purchase_order_updated_date,
+	concat((user_details1.jsonb -> 'personal'::text) ->> 'lastName'::text, ', ',
+              (user_details1.jsonb -> 'personal'::text) ->> 'firstName'::text)     AS  purchase_order_updated_by,
+    (o.jsonb -> 'metadata'::text) ->> 'updatedByUserId'::text AS purchase_order_updated_by_id,
+    o.jsonb ->> 'vendor'::text AS vendor_id,
+    organization_details.jsonb ->> 'code'::text AS vendor_code,
+    organization_details.jsonb ->> 'name'::text AS vendor_name,
+    o.jsonb ->> 'assignedTo'::text AS purchase_order_assigned_to_id,
+    concat((user_details4.jsonb -> 'personal'::text) ->> 'lastName'::text, ', ',
+              (user_details4.jsonb -> 'personal'::text) ->> 'firstName'::text) AS purchase_order_assigned_to,
+    jsonb_array_elements(o.jsonb -> 'notes'::text)::text AS purchase_order_notes,
+    ( SELECT array_to_string(array_agg(dept_id.value),',') AS array_agg
+           FROM jsonb_array_elements_text(po.jsonb -> 'acqUnitIds'::text) dept_id(value)) AS acqunit_ids,
+    ( SELECT array_agg(d.jsonb ->> 'name'::text) AS array_agg
+           FROM jsonb_array_elements_text(po.jsonb -> 'acqUnitIds'::text) dept_id(value)
+             JOIN src_acquisitions_unit d ON dept_id.value = d.id::text) AS acqunit_names
+   FROM src_purchase_order_line po
+     JOIN src_purchase_order o ON o.id = ((po.jsonb ->> 'purchaseOrderId'::text)::uuid)
+     LEFT JOIN src_users_users user_details ON user_details.id = (((o.jsonb -> 'metadata'::text) ->> 'createdByUserId'::text)::uuid)
+     LEFT JOIN src_users_users user_details1 ON user_details1.id = (((o.jsonb -> 'metadata'::text) ->> 'updatedByUserId'::text)::uuid)
+     LEFT JOIN src_users_users user_details2 ON user_details2.id = (((po.jsonb -> 'metadata'::text) ->> 'updatedByUserId'::text)::uuid)
+     LEFT JOIN src_users_users user_details3 ON user_details3.id = (((po.jsonb -> 'metadata'::text) ->> 'updatedByUserId'::text)::uuid)
+	 LEFT JOIN src_organizations ON organization_details.id = ((o.jsonb ->> 'vendor'::text)::uuid)
+     LEFT JOIN src_users_users user_details4 ON user_details4.id = ((o.jsonb ->> 'assignedTo'::text)::uuid)
+     LEFT JOIN LATERAL jsonb_array_elements(po.jsonb -> 'fundDistribution'::text) json_array_elements(value) ON true;
```

Comment:
What's going on with this join? It looks like there's some weird SQL voodoo going on (using `LATERAL`, 2 functions in a row with `jsonb_array_elements(...) json_array_elements(value)`, `ON true`). A comment here explaining what this is doing and why would be SUPER helpful.

And do we need the `ON true`? I'd think that would be implied, but with the other SQL voodoo, I'm not sure

Reply:
We are doing this because the fundDistribution field is an array of jsonb fundDistribution :[{code:1, value:1},{code:2, value:2}], so in order to get each code, value .... into a new row of text we are doing this messy sql code.
the use of ON TRUE in your query is likely necessary to satisfy the syntax otherwise it won't let me get the view being created successfully.